accounts-db/src/accounts_db.rs

```diff
@@ -12561,33 +12561,38 @@ pub mod tests {
 
     #[test]
     fn test_hash_stored_account() {
-        // This test uses some UNSAFE tricks to detect most of account's field
-        // addition and deletion without changing the hash code
+        // This test uses some UNSAFE tricks to detect most of hashing algo changes, resulting from
+        // account's field additions and deletions of StoredAccountMeta and AccountSharedData and
+        // hashing-order changes.
 
         const ACCOUNT_DATA_LEN: usize = 3;
-        // the type of InputTuple elements must not contain references;
+        // the type of InputFields elements must not contain references;
         // they should be simple scalars or data blobs
-        type InputTuple = (
+        #[repr(C)]
+        struct InputFields(
             Slot,
             StoredMeta,
             AccountMeta,
             [u8; ACCOUNT_DATA_LEN],
             usize, // for StoredAccountMeta::offset
             Hash,
         );
-        const INPUT_LEN: usize = std::mem::size_of::<InputTuple>();
+        const INPUT_LEN: usize = std::mem::size_of::<InputFields>();
         type InputBlob = [u8; INPUT_LEN];
         let mut blob: InputBlob = [0u8; INPUT_LEN];
 
-        // spray memory with decreasing counts so that, data layout can be detected.
+        // spray memory with decreasing integers so that, data layout change and/or hashing
+        // reordering can be detected.
         for (i, byte) in blob.iter_mut().enumerate() {
             *byte = (INPUT_LEN - i) as u8;
         }
 
-        //UNSAFE: forcibly cast the special byte pattern to actual account fields.
-        let (slot, meta, account_meta, data, offset, hash): InputTuple =
-            unsafe { std::mem::transmute::<InputBlob, InputTuple>(blob) };
+        //UNSAFE: forcibly slice off actual fields from the special byte pattern.
```

Comment:
Why the rewording of "cast" to "slice off"? This is still transmuting the entire thing, not just a slice of it, right?

Reply:
no particular reason other then i was thinking i've improve my english writing skill over the 3 years since I wrote the original comment. :) seems confusing; just reverted: 99ec8d53fc2